src/middlewared/middlewared/plugins/reporting/events.py

```diff
@@ -63,7 +64,18 @@ def run_sync(self):
 
         while not self._cancel_sync.is_set():
             # this gathers the most recent metric recorded via netdata (for all charts)
-            netdata_metrics = self.middleware.call_sync('netdata.get_all_metrics')
+            retries = 2
+            while retries > 0:
+                try:
+                    netdata_metrics = self.middleware.call_sync('netdata.get_all_metrics')
+                except Exception:
+                    retries -= 1
+                    if retries <= 0:
+                        raise
+
+                    time.sleep(random.randint(0, interval))
```

Comment:
Why sleep for 0 seconds?

Reply:
agree, we don't need to do `random.randint(0, interval)`. We just need to sleep for arbitrary number each time, maybe 0.5 seconds